src/psyclone/tests/domain/nemo/transformations/nemo_arrayrange2loop_trans_test.py

```diff
@@ -617,3 +617,81 @@ def test_validate_not_outermost_range():
     assert ("Error in NemoArrayRange2LoopTrans transformation. This "
             "transformation can only be applied to the outermost "
             "Range." in str(info.value))
+
+
+def test_character_validation(fortran_reader):
+    '''Check that the validate method returns an exception if the
+    lhs of the assignment contains a character array and the allow_string
+    option isn't defined, and that it doesn't return an exception if the
+    allow_string option is True.'''
+
+    code = '''subroutine test()
+    character :: a(100)
+    character :: b(100)
```

Comment:
Added this as a separate check because I think it is a little cleaner.

Reply:
Yeah - we don't support `LEN=xxx`. (We only support single characters I think - I can't remember why we have this limitation but it's probably because we included CHARACTER type in the PSyIR type system but then didn't actually need it for anything.)